### ISSUE TITLE
test_normalize_url_slashes and test_normalize_url_with_base_unc_path fail with Python 3.13

### ISSUE THREAD

**Issue by glaubitz** [sissaschool/xmlschema#408]
On Python 3.13, the tests `tests test_normalize_url_slashes` and `test_normalize_url_with_base_unc_path` fail with:

```
[   44s] =================================== FAILURES ===================================
[   44s] ___________________ TestLocations.test_normalize_url_slashes ___________________
[   44s] 
[   44s] self = <tests.test_locations.TestLocations testMethod=test_normalize_url_slashes>
[   44s] 
[   44s]     def test_normalize_url_slashes(self):
[   44s]         # Issue #116
[   44s]         url = '//anaconda/envs/testenv/lib/python3.6/site-packages/xmlschema/validators/schemas/'
[   44s]         if os.name == 'posix':
[   44s]             normalize_url(url)
[   44s]             self.assertEqual(normalize_url(url), pathlib.PurePath(url).as_uri())
[   44s]         else:
[   44s]             # On Windows // is interpreted as a network share UNC path
[   44s]             self.assertEqual(os.name, 'nt')
[   44s]             self.assertEqual(normalize_url(url),
[   44s]                              pathlib.PurePath(url).as_uri().replace('file://', 'file:////'))
[   44s]     
[   44s]         self.assertRegex(normalize_url('/root/dir1/schema.xsd'),
[   44s]                          f'file://{DRIVE_REGEX}/root/dir1/schema.xsd')
[   44s]     
[   44s] >       self.assertRegex(normalize_url('////root/dir1/schema.xsd'),
[   44s]                          f'file://{DRIVE_REGEX}//root/dir1/schema.xsd')
[   44s] E       AssertionError: Regex didn't match: 'file:////root/dir1/schema.xsd' not found in 'file://////root/dir1/schema.xsd'
[   44s] 
[   44s] tests/test_locations.py:314: AssertionError
[   44s] _____________ TestLocations.test_normalize_url_with_base_unc_path ______________
[   44s] 
[   44s] self = <tests.test_locations.TestLocations testMethod=test_normalize_url_with_base_unc_path>
[   44s] 
[   44s]     def test_normalize_url_with_base_unc_path(self,):
[   44s]         base_unc_path = '\\\\filer01\\MY_HOME\\'
[   44s]         base_url = PureWindowsPath(base_unc_path).as_uri()
[   44s]         self.assertEqual(str(PureWindowsPath(base_unc_path)), base_unc_path)
[   44s]         self.assertEqual(base_url, 'file://filer01/MY_HOME/')
[   44s]     
[   44s]         # Same UNC path as URI with the host inserted in path
[   44s]         base_url_host_in_path = base_url.replace('file://', 'file:////')
[   44s]         self.assertEqual(base_url_host_in_path, 'file:////filer01/MY_HOME/')
[   44s]     
[   44s]         self.assertEqual(normalize_url(base_unc_path), base_url_host_in_path)
[   44s]     
[   44s]         with patch.object(os, 'name', 'nt'):
[   44s]             self.assertEqual(os.name, 'nt')
[   44s]             path = PurePath('dir/file')
[   44s]             self.assertIs(path.__class__, PureWindowsPath)
[   44s]     
[   44s]             url = normalize_url(r'dev\XMLSCHEMA\test.xsd', base_url=base_unc_path)
[   44s]             self.assertEqual(url, 'file:////filer01/MY_HOME/dev/XMLSCHEMA/test.xsd')
[   44s]     
[   44s]             url = normalize_url(r'dev\XMLSCHEMA\test.xsd', base_url=base_url)
[   44s]             self.assertEqual(url, 'file:////filer01/MY_HOME/dev/XMLSCHEMA/test.xsd')
[   44s]     
[   44s]             url = normalize_url(r'dev\XMLSCHEMA\test.xsd', base_url=base_url_host_in_path)
[   44s] >           self.assertEqual(url, 'file:////filer01/MY_HOME/dev/XMLSCHEMA/test.xsd')
[   44s] E           AssertionError: 'file://///filer01/MY_HOME/dev/XMLSCHEMA/test.xsd' != 'file:////filer01/MY_HOME/dev/XMLSCHEMA/test.xsd'
[   44s] E           - file://///filer01/MY_HOME/dev/XMLSCHEMA/test.xsd
[   44s] E           ?      -
[   44s] E           + file:////filer01/MY_HOME/dev/XMLSCHEMA/test.xsd
[   44s] 
[   44s] tests/test_locations.py:283: AssertionError
[   44s] =========================== short test summary info ============================
[   44s] FAILED tests/test_locations.py::TestLocations::test_normalize_url_slashes - A...
[   44s] FAILED tests/test_locations.py::TestLocations::test_normalize_url_with_base_unc_path
[   44s] ================= 2 failed, 1245 passed, 115 skipped in 15.54s =================
```

**Comment by brunato** [comment]
A fix for this is available with v3.3.2. The problem is the same of #405 (a change in `urllib.parse.urlunsplit` included in Python 3.12.4 and then in the following beta release of Python 3.13).

Thank you

**Comment by glaubitz** [comment]
> A fix for this is available with v3.3.2. The problem is the same of #405 (a change in `urllib.parse.urlunsplit` included in Python 3.12.4 and then in the following beta release of Python 3.13).
> 
> Thank you

Thanks a lot! I already updated the `python-xmlschema` package in openSUSE to 3.3.2.
